app/controllers/patients_controller.rb

```diff
@@ -60,6 +60,10 @@ def edit
   end
 
   def update
+    if @patient.update_attributes params[:pledge_sent]
```

Comment:
Very slight adjustment, but can you move this block so that it takes place within the `update_attributes patient_params` thing on L67? Something like this should work:

```
if @patient.update_attributes patient_params
  if @patient.previous_changes[:pledge_sent] && @patient.pledge_sent
    @patient.update_attributes pledge_sent_at: Time.zone.now, pledge_sent_by: current_user
  end
  @patient.reload
...
```

I know it's kind of a pain and I'm sorry to ask when this is working great, but this is probably our highest traffic endpoint, so keeping the db calls to a minimum is pretty clutch here. Happy to handle it if you would prefer to get pluggin' away on something else.